lib/usd/ui/factories/TreeModelFactory.h

```diff
@@ -0,0 +1,145 @@
+//
+// Copyright 2019 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#pragma once
+
+#include "../api.h"
+
+#include <memory>
+#include <unordered_set>
+#include <string>
+#include <vector>
+
+#include <QtCore/QList>
+
+#include <pxr/usd/usd/prim.h>
+#include <pxr/usd/usd/stage.h>
+
+class QObject;
+class QStandardItem;
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+MAYAUSD_NS_DEF {
+
+class TreeModel;
+
+/**
+ * \brief Factory to create a tree-like structure of USD content suitable to be displayed in a TreeView.
+ */
+class MAYAUSD_UI_PUBLIC TreeModelFactory
+{
+private:
+	/**
+	 * \brief Default Constructor (deleted).
+	 */
+	TreeModelFactory() = delete;
+
+public:
+	/**
+	 * \brief Create an empty TreeModel.
+	 * \param parent A reference to the parent of the TreeModel.
+	 * \return An empty TreeModel.
+	 */
+	static std::unique_ptr<TreeModel> createEmptyTreeModel(QObject* parent = nullptr);
+
+	/**
+	 * \brief Create a TreeModel from the given USD Stage.
+	 * \param stage A reference to the USD Stage from which to create a TreeModel.
+	 * \param parent A reference to the parent of the TreeModel.
+	 * \return A TreeModel created from the given USD Stage.
+	 */
+	static std::unique_ptr<TreeModel> createFromStage(const UsdStageRefPtr& stage, QObject* parent = nullptr, int* nbItems = nullptr);
```

Comment:
I see how an argument could be made for size_t. I made it an int because it is eventually used in USDImportDialog.cpp as an arg to a QString (to replace the %1). QString arg method uses int. So I just used int. If I switched it to size_t from the method then I'll have to cast to an int for the QString. What do you think?

I did forget to add to the doc string. I'll do that.